task1-gradient-descent/GradientDescent.ipynb

```diff
@@ -1 +1,1430 @@
dvEbir+uQyORYTp6PNjGBlL+yb6iTo5sj4YGv12tEUDjm1haYWZplGK6iVqmp7FGVgQuGokpNY9uSLQCc3HGc6s1rsu7yz8hkMm6euc61Y1eMnj04LhkXS1P9tLOVKfGpKhJS1fqhcn5RCTiYmzD9yG0ehcVhZaJgZG3dsDKJRIJcImHSwZscfRxC/cLO5LfLuH8zFomVHZq411eitHFRSEzMQWmqHyonsbBF8/wBaef2oI0IRF7JA5M2Q0jePAv13XPIi1bErP93IJWh9ruL+tmtbMsvdXZGHfZ626MJC0NqaYnE3Fw/VE4THExq8Ottn9WQIaScPw8qFWn37mHWuDFpt28jUSoxrVMHbTYO03JwcyTijbZv8S9tf9CCYaSlprF1yW8APL7xiDrt6vPg6n0USgVVm9VAnWb8ZZDaOqKJep1dGx2OxMwCTM0MhsqZdhpC6vlDaAJ9Dd/vnBuN3yPMBs1AYm2P+tldUvb+bPTc6X3O6z9HfGH3DGXbacF79+5RvHhxfUfolbp162Jvb09oaCht2rShadOmVK1alWXLlrFy5UpcXV0z/bxmzZpx+PBhAG7dukWxYsVQKLLvLNt/hUQqMTiz+YpGnfkXQWFmQsfVw7HP58q+8RuMHS8jiQQyxjV4DKSyfmtQq0k7cyj7cr2NRKK7EvKGDOv4LXVPtpzk6uRNqJNSSItN5P66g+RtVgmpQoZVPmfS4pM40mYmZwatotL0bti75zfOskgzz5h+/SvqtQKNmrRzh7P+mK+KYDFuCakn9qK6dckYSTMlkUgzj/+W9t5p1Qjs87uw29uwvecqXQDP7VO5uPEID49fN0ZcQxLpv7b9V9IunyW6ZxuStv2C1dRFIJGQ8vdfJP6wHFKS0SbGk7xvO8pq2Xv1WpJF+8lq/V85com+5buzfenvTP51OhKJhI4jOxEbGYNnxZ4MrNYHS1srWnoaf5iiFu2bI24BkKXbpak0Ws75htHePS9butagU/mvGLbnGqmq18s3p1lZTgxsQGxyGusvPjF6bj2JlEwbULr2o40NJ2X392gjdFcAVFePILFxQmLtiKJ6K7SJcSStHUPS+nFITC2QV2ycTeHJcvuozewRwKam2Eyfjix3bmIXLgQgbs0aABx++AGb2bNJuXo1W+9ZkUr/v23PlSOX6FO+GzuW/s6UX2cgkUjYOPsntFotCw8sY9yGidw6cwNVdhyMSyT8W9tR1GoOajWqi39nfLtUjqx4eZJ+nk/iolFIzK0waZlxeKwxfdbrPydoNMb5+URl25UhqVSKiYlJlq+/Gian0WiYP38+T58+pWbNrC8/NmjQgGXLlqHRaDh48CDNmjXjwIEDxoj+n1N/dHuKNdKN9TaxMiPkwXP9a1au9iRFx5OWlJLhfTa5HOj84xjCnwTyS6fZ2XLD+Js0kaEoCpXQT0vsHNHExxrcBKyo5YFEaYLljLW6IR1KJZYz1pKwdBLa6IjMPvajK+PVnjweFXR5LM2ITreOzV3tSImKR/3GOk58EYFjhUKZ1hVoX5Ooe/5E39d9jkQCGpWaxOBoAJ5uPQ1AvG8IoZcf4lC+EJG3fT/6cmkjwpAUKK6fltg6ok0wXP/KGo1BaYrF1DUgk4NSicXUNSQun4Q2JhJ55XqYdR1K0pZVqC4bf6hEw1EdKNFY939hYmlOyEN//WvWrvYkvqW9d/9xLGFPAvnxjfbu3qo6rWf15s+pv3Br33mjLwOAJjwEedHXbV/q4IgmLhZSXq97qWtupHb2qO7rrl6lHDuA+YDRSCytUFSohtr3CWq/l1dLJZJsORj8dnQXKjWqDICZlTn+D/z0r9m7OhAfHUfKG+vfNZ8rtk52PLh6H4AT24/Rf+4gLGwsqdK0Gj9N24AqTYUqTcWpncep1rwGf23Ya9TlcLUy43ZQjH46ND4FaxMFZorXu1EnSxMK2Fvi7mYLQP1CLsz8+w4BMYkExyVR2NEKZ0tTzJVymhZz49jjEKNmTk8bF4HUtYB+WmJpizY5AVSvH14iccyN1Ckv6vsXX79RIgGNGlnhCqSe+F037j81CdXdC8iLVkR1LePBrzFoQkNRlEjX/h0d0cTGQrLhAyCkzs7Yzp2L2t+fqJEjIVW3fBJzc+LWrkUbpxvyZN61K6qXQ+6MRdf2q+h+XyZtPy7Ttu+GrZOtvu0f334Uz5dt39TchM1zfyE+RjcMut2QjvphX8akiQpDnr+Yflpi44A2IQ5SX2dXVG0EChPMx30PcjkolJiP+56ktdPRxEagunlefxUp7coJTJp2Nnru/8r6F4wv264MlS5dmnv37mW4CrFkyRIuXXp9ZlgqlTJu3DhCQkL48ccfs/y8V0Plrl27xsWLFz+ZIXKfgxNLdrG2+UTWNp/ID22nkad8Yezz6+7bqNS1IQ+OXMvwHqWFKb22Teb+oavsHLYyRzpCAKo715AVLIHUJbcuV/1WqK4bHowmzBpK/BRP4qcNJGHpREhNJX7awGzrCAHcWrhL/6CDQy2n41ihMFYFdOu4SI+GBBzJOIQy8NTtLOtsi+ehrFd7JFIJMlMFxXp74Lf3IgnPw4i45UPBb3Rn+E0drXGqVITIm8Z5yp/q3sv175wLAGXdlqTduGBQkzB3OAnT+5MwcxCJ30+G1FQSZg7SdYTKVMO002ASlk7Ilo4QwLGlO1nZfCIrm09k7ddTyVuuCA75dVecq3RtyP2/M2/v/bZO4d6hK2wbtsKgvRdvWIGW03rwS/d52dYRAki7cQV50ZJI3XRt36RJa9IuGz41UGrngOXoqUisbHTLUacxan8ftHGxyL4qgFnnPrp7P5RKTJt/Teo54/8fbFuyBa/mo/BqPoqJbcdRpHwxXPO7AeDRtSlXjlzO8B5bZ3tGrhyLlZ0VALXa1sX/oT/x0XH43HlGjRa6E2UyuYxKjavw6PrDDJ/xsVXP58Dt4Gj8ohIA2HnLn3qFnA1qauZ34kVMEvdCdJ2mawGRSJCQ28aMI4+CWX/xCVq
```

Comment:
<p>See this video for the logic behind:</p><p>https://www.youtube.com/watch?v=CbsegCURCd0</p><p> </p><p>The answer is correct but I'm not sure if the gist behind is understood here. This is a good explanation:</p><p>https://stackoverflow.com/questions/49147774/what-is-random-state-in-sklearn-model-selection-train-test-split-example#:~:text=If%20you%20don%27t,and%20test%20datasets.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bogaziciai/assignment/pull/14/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>